tests/component_derive_test.rs

```diff
@@ -1112,6 +1112,24 @@ fn derive_struct_component_field_type_override() {
     }
 }
 
+#[test]
+fn derive_struct_component_field_type_path_override() {
+    let post = api_doc! {
+        struct Post {
+            id: i32,
+            #[component(value_type = path::to::Foo)]
+            value: i64,
+        }
+    };
+
+    let component_ref: &str = post
+        .pointer("/properties/value/$ref")
+        .unwrap()
+        .as_str()
+        .unwrap();
+    assert_eq!(component_ref, "#/components/schemas/path::to::Foo");
```

Comment:
I guess we need to think this namespacing over again. This kind of a name `"#/components/schemas/path::to::Foo");` is not valid OpenAPI spec name and will cause errors when rendered. 

The name should be just single word.

Reply:
Sorry, it works fine in Swagger, and RapiPDF, and I can't find anything in the spec yet which says that the name cannot contain a `::`. Where is the reference for this?

Reply:
From what I could find out openapi 3.0 uses the json pointer specification for these references: https://datatracker.ietf.org/doc/html/draft-ietf-appsawg-json-pointer-04 there is also no mention in here about colons being an illegal character.

Reply:
Sorry forgot to add the details previously. Added them now down there :point_down: